tests/unit/submissionworker/test_worker.py

```diff
@@ -0,0 +1,189 @@
+import boto3
+import os
+import shutil
+import tempfile
+import zipfile
+
+from datetime import timedelta
+from moto import mock_sqs
+from os.path import join
+
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.contrib.auth.models import User
+from django.utils import timezone
+
+from allauth.account.models import EmailAddress
+from rest_framework.test import APITestCase, APIClient
+
+from challenges.models import (
+    Challenge,
+    ChallengePhase,
+)
+from participants.models import Participant, ParticipantTeam
+from hosts.models import ChallengeHostTeam
+from jobs.models import Submission
+
+from scripts.workers.submission_worker import (
+    download_and_extract_file,
+    create_dir,
+    create_dir_as_python_package,
+    return_file_url_per_environment,
+    get_or_create_sqs_queue
+)
+
+
+class BaseAPITestClass(APITestCase):
+
+    def setUp(self):
+
+        self.client = APIClient(enforce_csrf_checks=True)
+
+        self.user = User.objects.create(
+            username='someuser',
+            email="user@test.com",
+            password='secret_password')
+
+        EmailAddress.objects.create(
+            user=self.user,
+            email='user@test.com',
+            primary=True,
+            verified=True)
+
+        self.user1 = User.objects.create(
+            username='someuser1',
+            email="user1@test.com",
+            password='secret_password1')
+
+        EmailAddress.objects.create(
+            user=self.user1,
+            email='user1@test.com',
+            primary=True,
+            verified=True)
+
+        self.client.force_authenticate(user=self.user1)
+
+        self.challenge_host_team = ChallengeHostTeam.objects.create(
+            team_name='Test Challenge Host Team',
+            created_by=self.user)
+
+        self.participant_team = ParticipantTeam.objects.create(
+            team_name='Participant Team for Challenge',
+            created_by=self.user1)
+
+        self.participant = Participant.objects.create(
+            user=self.user1,
+            status=Participant.SELF,
+            team=self.participant_team)
+
+        self.BASE_TEMP_DIR = tempfile.mkdtemp()
+
+        f = open(join(self.BASE_TEMP_DIR, 'dummy_input.txt'), "x")
+        f = open(join(self.BASE_TEMP_DIR, 'dummy_input.txt'), "w")
+        f.write("file_content")
+
+        self.zipped = zipfile.ZipFile(join(self.BASE_TEMP_DIR, 'test_zip.zip'), 'w')
+        self.zipped.write(join(self.BASE_TEMP_DIR, 'dummy_input.txt'), 'dummy_input.txt')
+        self.zipped.close()
+        file = open(join(self.BASE_TEMP_DIR, 'test_zip.zip'), 'rb')
+        self.z = SimpleUploadedFile(join(self.BASE_TEMP_DIR, 'test_zip.zip'), file.read(), content_type='application/zip')
```

Comment:
@KhalidRmb please name the variable properly

Reply:
Actually, we don't need this part since we're skipping the download_and_extract_zip_file test. So, I'll go ahead and remove this for using later.